test/unit/node-gyp-build-resolve/output.js

```diff
@@ -0,0 +1,5133 @@
+    ffi_type.alignment = 0;
+    ffi_type.type = 13; // FFI_TYPE_STRUCT
+
+    // first we have to figure out the number of "elements" that will go in the
+    // array. this would normally just be "numFields" but we also have to account
+    // for arrays, which each act as their own element
+    for (let i = 0; i < numFields; i++) {
+      field = fields[fieldNames[i]];
+      if (field.type.fixedLength > 0) {
+        // a fixed-length "ref-array" type
+        numElements += field.type.fixedLength;
+      } else {
+        numElements += 1;
+      }
+    }
+
+    // hand-crafting a null-terminated array here.
+    // XXX: use "ref-array"?
+    const size = ref.sizeof.pointer * (numElements + 1); // +1 because of the NULL terminator
+    const elements = ffi_type.elements = Buffer.alloc(size);
+    let index = 0;
+    for (let i = 0; i < numFields; i++) {
+      field = fields[fieldNames[i]];
+      if (field.type.fixedLength > 0) {
+        // a fixed-length "ref-array" type
+        ffi_type_ptr = Type(field.type.type);
+        for (var j = 0; j < field.type.fixedLength; j++) {
+          elements.writePointer(ffi_type_ptr, (index++) * ref.sizeof.pointer);
+        }
+      } else {
+        ffi_type_ptr = Type(field.type);
+        elements.writePointer(ffi_type_ptr, (index++) * ref.sizeof.pointer);
+      }
+    }
+    // final NULL pointer to terminate the Array
+    elements.writePointer(ref.NULL, index * ref.sizeof.pointer);
+    // also set the `ffi_type` property to that it's cached for next time
+    ret = type.ffi_type = ffi_type.ref();
+  }
+
+  if (!ret && type.name) {
+    // handle "ref" types other than the set that node-ffi is using (i.e.
+    // a separate copy)
+    if ('CString' == type.name) {
+      ret = type.ffi_type = bindings.FFI_TYPES.pointer;
+    } else {
+      let cur = type;
+      while (!ret && cur) {
+        ret = cur.ffi_type = bindings.FFI_TYPES[cur.name];
+        cur = Object.getPrototypeOf(cur);
+      }
+    }
+  }
+
+  assert(ret, 'Could not determine the `ffi_type` instance for type: ' + (type.name || type))
+  debug('returning `ffi_type`', ret.name)
+  return ret;
+}
+
+module.exports = Type;
+
+
+/***/ }),
+
+/***/ 714:
+/***/ ((module) => {
+
+"use strict";
+
+
+module.exports = (flag, argv = process.argv) => {
+	const prefix = flag.startsWith('-') ? '' : (flag.length === 1 ? '-' : '--');
+	const position = argv.indexOf(prefix + flag);
+	const terminatorPosition = argv.indexOf('--');
+	return position !== -1 && (terminatorPosition === -1 || position < terminatorPosition);
+};
+
+
+/***/ }),
+
+/***/ 542:
+/***/ ((module, exports, __webpack_require__) => {
+
+"use strict";
+
+const assert = __webpack_require__(357);
+const inspect = __webpack_require__(669).inspect;
+const debug = __webpack_require__(240)('ref');
+const os = __webpack_require__(87);
+const path = __webpack_require__(622);
+
+exports = module.exports = require(__webpack_require__.ab + "prebuilds/darwin-x64/node.napi.node");
```

Comment:
Looks like this test is failing on windows and linux because this is the result when running on macos.

You might need to do a find/replace on the output here before running the test to make sure it works on all platforms.